lib/event_store_client/data_decryptor.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module EventStoreClient
+  class DataDecryptor
+    def call
+      return encrypted_data if encryption_metadata.empty?
+
+      key = key_repository.find(encryption_metadata[:key])
```

Comment:
find will raise an error when not found no need to handle it?

Reply:
I don't expect it to be `ActiveRecord` object. But yes, I don't want to bother about the details of the implementation - so I don't care if find returns nil or rises the error. In both cases I should rise my own, controlled error.

https://github.com/yousty/encryptor/blob/master/app/repositories/encryption_key_repository.rb#L6